pkg/downloader/downloader.go

```diff
@@ -0,0 +1,195 @@
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/util/log"
+)
+
+//DownloadConfigs get configurations from Dynatrace API and transforms them into objects
+func DownloadConfigs(env manifest.EnvironmentDefinition, listApis map[string]api.Api, projectId string) (projectv2.ConfigsPerApisPerEnvironments, []error) {
+	var errors []error
+	result := make(projectv2.ConfigsPerApisPerEnvironments)
+	result[env.Name] = make(map[string][]configv2.Config)
+
+	token, err := env.GetToken()
+	if err != nil {
+		log.Error("error retrieving token for enviroment %v %v", env.Name, err)
+		errors = append(errors, err)
+		return nil, errors
+	}
+	client, err := rest.NewDynatraceClient(env.Url, token)
+	if err != nil {
+		log.Error("error creating dynatrace client for enviroment %v %v", env.Name, err)
+		errors = append(errors, err)
+		return nil, errors
+	}
+	for _, api := range listApis {
+		creator := dynatraceparser.NewDynatraceParser()
+		configs, errs := createConfigforTargetApi(env, creator, client, api, projectId)
+		//apiconfig, err := createConfigFromApiV2(env, api, projectId, token, client)
+		if len(errs) > 0 {
+			log.Error("error getting configs for env %v %v", env.Name, errs[0])
+			errors = append(errors, errs[0])
+			continue
+		}
+		result[env.Name][api.GetId()] = append(result[env.Name][api.GetId()], configs...)
+	}
+	result[env.Name] = solveDependencies(projectId, result[env.Name])
+	if errors != nil {
+		return nil, errors
+	}
+	return result, nil
+}
+func GetDependencies() map[string][]string {
+	dependenciesPerEnvironment := make(map[string][]string)
+	return dependenciesPerEnvironment
+}
+func solveDependencies(projectId string, configs map[string][]configv2.Config) map[string][]configv2.Config {
+	//temp object to hold the ids and coordinates
+	tempConfig := make(map[string]coordinate.Coordinate)
+	//builds temp object
+	for _, configType := range configs {
+		for _, config := range configType {
+			tempConfig[config.Coordinate.DynatraceId] = config.Coordinate
+		}
+	}
+	//Brute force for now, TODO find an optimize way to explore files
+	for _, configType := range configs {
+		for _, config := range configType {
+			content := config.Template.Content()
+			for key, val := range tempConfig {
+				exists := strings.Contains(content, key)
+				if exists {
+					log.Debug(val.Api)
+					//WARNING: validate multiple dependencies in 1 config!!
+					//should update template and coordinate
+					config = newDependencyConfig(projectId, config, val)
+				}
+			}
+		}
+	}
+	return configs
+}
+func newDependencyConfig(projectId string, config configv2.Config, dependencyCoor coordinate.Coordinate) configv2.Config {
+	newContent := strings.ReplaceAll(config.Template.Content(), dependencyCoor.DynatraceId, dependencyCoor.Config)
+	idProperty := "id" //since all depends on id to relative configs
+	ref := refParam.New(dependencyCoor.Project, dependencyCoor.Api, dependencyCoor.Config, idProperty)
+	config.Template.UpdateContent(newContent)
```

Comment:
would personally rather create a new Template with the newContent here, than extending it with an `UpdateContent` 
generally it's not something that will be modified, so I don't think we should make it possible just for this use